### PR TITLE
dep: update nix dependency

### PR DESCRIPTION
**Pull request by bergwolf**
To fix CVE-2021-45707 that affects nix < 0.20.2.

Fixes: #4929

### CODE REVIEW

File: src/agent/rustjail/Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ serde_derive = "1.0.91"
 oci = { path = "../../libs/oci" }
 protocols = { path ="../../libs/protocols" }
 caps = "0.5.0"
-nix = "0.23.0"
+nix = "0.24.1"
```
Comment (liubin) [inline src/agent/rustjail/Cargo.toml:14]: Why not use a more higher version, `0.24.2` or `0.25.0`?

Comment (bergwolf) [reply src/agent/rustjail/Cargo.toml:14]: There are quite a few breaking API changes...

**Review by snir911 — APPROVED** [review]

**Review by liubin — APPROVED** [review]

**Pull request merged**